lib/layers/presentation/sing_in_page/sing_in_page.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:wot_statistic/layers/domain/entities/user.dart';
+import 'package:wot_statistic/layers/presentation/settings_page/settings_page.dart';
+import 'package:wot_statistic/layers/presentation/sing_in_page/bloc/sing_in_cubit.dart';
+import 'package:wot_statistic/layers/presentation/sing_in_page/widgets/animated_background.dart';
+import 'package:wot_statistic/layers/presentation/sing_in_page/widgets/region_picker.dart';
+import 'package:wot_statistic/layers/presentation/sing_in_page/widgets/themed_button.dart';
+import 'package:wot_statistic/layers/presentation/sing_in_page/widgets/user_picker.dart';
+import 'package:wot_statistic/layers/presentation/singup_user/singup_user_page.dart';
+
+import '../../../common/theme/text_styles.dart';
+import '../statistic_page/statistic_page.dart';
+
+class SingInPage extends StatelessWidget {
+  const SingInPage({Key? key}) : super(key: key);
+
+  static const String id = "WOT Statistic";
+
+  @override
+  Widget build(BuildContext context) {
+    final SingInCubit cubit = context.read<SingInCubit>();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(id, style: appBarTitle(context)),
+        actions: [
+          IconButton(
+            onPressed: () {
+              //todo search player in directed realm (cubit)
+            },
+            icon: const Icon(Icons.search),
+          ),
+          IconButton(
+            onPressed: () {
+              Navigator.of(context).pushNamed(SettingsPage.id);
+            },
+            icon: const Icon(Icons.settings),
+          )
+        ],
+      ),
+      body: BlocConsumer<SingInCubit, SingInState>(
+        listener: (prevState, currentState) {
+          if (currentState.status == SingInStatus.error) {
+            ScaffoldMessenger.of(context).showSnackBar(
+              SnackBar(
+                content: Text(
+                  currentState.errorMessage!,
```

Comment:
Please avoid bang. You can create subclasses like SuccessSignInState and ErrorSignInState. Check if currentState is instance of some of it, and then use fields from it. ALso it will help to not extend model with field which are not need, like SuccessSignInState will not need errorMessage